lib/jsonapi/include_directive.rb

```diff
@@ -40,6 +42,28 @@ def [](key)
       end
     end
 
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge(other)
+      fail ArgumentError,
+           "the value of 'other' MUST be an IncludeDirective" unless
+          other.is_a?(IncludeDirective)
+
+      hash = to_hash.dup
+      Parser.deep_merge!(hash, other.to_hash)
+
+      IncludeDirective.new(hash, options.merge!(other.options))
+    end
+
+    # @param another_directive [IncludeDirective]
+    # @return [IncludeDirective]
+    def merge!(other)
+      new = merge(other)
```

Comment:
I'd stay clear from `new` as a local variable name as it throws off syntax highlighting, and might confuse readers.